elide-core/src/main/java/com/yahoo/elide/security/permissions/expressions/AnyFieldExpression.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2015, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.security.permissions.expressions;
+
+import com.yahoo.elide.security.permissions.ExpressionResult;
+
+import static com.yahoo.elide.security.permissions.ExpressionResult.FAIL;
+import static com.yahoo.elide.security.permissions.ExpressionResult.PASS;
+
+/**
+ * Implementation of joining expression results by any field success or entity success.
+ */
+public class AnyFieldExpression implements Expression {
+    private final Expression entityExpression;
+    private final Expression fieldExpression;
+
+    public AnyFieldExpression(final Expression entityExpression, final Expression fieldExpression) {
+        this.entityExpression = entityExpression;
+        this.fieldExpression = fieldExpression;
+    }
+
+    @Override
+    public ExpressionResult evaluate() {
+        ExpressionResult fieldResult = (fieldExpression == null) ? FAIL : fieldExpression.evaluate();
+        if (fieldResult != ExpressionResult.FAIL) {
+            return fieldResult;
+        }
+        return (entityExpression == null) ? PASS : entityExpression.evaluate();
```

Comment:
Yes it would. I think there is a question as to what the default "correct" logic for this case is.

In general, if nothing is specified, we consider a bean to be open, correct? In this case, we should ultimately return PASS since this is an `AnyFieldExpression` which determines whether there is access either on a specific field or the entire entity.

I do not believe this poses any real security risks, however, since `SpecificFieldExpressions` will be run for all the fields as they are serialized and before they are updated. I believe this is for other optimizations.